src/endpoints/tokens/token.service.ts

```diff
@@ -173,6 +113,33 @@ export class TokenService {
       tokens = tokens.filter(token => identifierArray.includes(token.identifier.toLowerCase()));
     }
 
+    if (filter.sort) {
+      const order = filter.order ?? SortOrder.desc;
+
+      let criteria: (token: Token) => number;
+
```

Comment:
I would extract each switch into separate method

Reply:
extracted into a single, unified function for sorting